foca/api/register_openapi.py

```diff
@@ -0,0 +1,136 @@
+"""Functions for registering OpenAPI specs with a Connexion app instance."""
+
+from json import load, decoder
+import logging
+import os
+from typing import Dict, List, Optional
+
+from connexion import App
+from yaml import safe_load, safe_dump, parser
+
+from foca.config.config_parser import get_conf
+from foca.errors.errors import InvalidSpecification
+
+
+# Get logger instance
+logger = logging.getLogger(__name__)
+
+
+class OpenAPIConfig(Dict):
+    """Helper class for the configuration parameters of the specification files.
+    Inherits from the Dictionary class.
+
+    Attributes (optional):
+        out_file: The directory for the output specification file, in case it's
+                either modified or in JSON format
+        append: Fields to be added/modified in the root of the specification file
+        add_operation_fields: Fields to be added/modified in Operation Objects
+    """
+
+    def __init__(
+            self,
+            out_file: Optional[str],
+            append: Optional[List[Dict]],
+            add_operation_fields: Optional[Dict]
+    ) -> None:
+        super().__init__()
+        if out_file:
+            self.out_file = out_file
+        if append:
+            self.append = append
+        if add_operation_fields:
+            self.add_operation_fields = add_operation_fields
+
+
+def register_openapi(
+        app: App,
+        specs_in: Dict[str, OpenAPIConfig]
+) -> App:
+    """
+    Registers OpenAPI specs with Connexion app
+
+    Args:
+        app: A Connexion app instance
+        specs_in: A Dictionary containing the path of configuration

```

Comment:
`_absolute_ paths` instead of `paths` and `OpenAPI specification` instead of `configuration` files